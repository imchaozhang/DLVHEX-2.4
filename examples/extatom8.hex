d(c1).
d(c2).
d(c3).
d(c4).

% guess for c1 p/1 or q/1 depending on r/1
p(c1) :- r(X), not q(c1).
q(c1) :- r(X), not p(c1).

% create s/1 for all d/1
s(X)  :- d(X), not p(c1).
s(X)  :- d(X), not q(c1).

% create r/1 for all X
r(X)  :- d(X), s(X), not &testC[t](X).

