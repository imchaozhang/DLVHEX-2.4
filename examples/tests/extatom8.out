{p(c1),s(c1),s(c2),s(c3),s(c4),r(c1),r(c2),r(c3),r(c4)}
{q(c1),s(c1),s(c2),s(c3),s(c4),r(c1),r(c2),r(c3),r(c4)}
